snapcraft/commands/remote.py

```diff
@@ -88,15 +89,58 @@ def fill_parser(self, parser: argparse.ArgumentParser) -> None:
             help="acknowledge that uploaded code will be publicly available.",
         )
 
+    def _get_base(self) -> str:
+        """Get a valid base from the project's snapcraft.yaml.
```

Comment:
is this the base or the effective base? As in, what should happen for the following

    build-base: devel
    base: core24

or

    build-base: core22
    base: core18

or

    build-base: core22
    base: bare

Reply:
It's the effective base, so I think:


`devel` for
```
build-base: devel
base: core24
```

`core22` for 
```
build-base: core22
base: core18
```

`core22` for 
```
build-base: core22
base: bare
```

Is that OK?  If so, I can update the function names, docstrings, and add tests.

Reply:
`_get_effective_base()` seems to be the natural choice in this context.